CRM/Appearancemodifier/Form/Petition.php

```diff
@@ -184,4 +142,48 @@ private function getPetition(int $id): array
         }
         return $petition['values'][0];
     }
+
+    /*
+     * This function gathers the consent custom fields that
+     * are present in this petition form.
+     */
+    protected function consentActivityCustomFields(): void
+    {
+        // gather the custom fields from the service.
+        $consentActivityConfig = new CRM_Consentactivity_Config('consentactivity');
+        $consentActivityConfig->load();
+        $config = $consentActivityConfig->get();
+        if (array_key_exists('custom-field-map', $config)) {
+            $map = $config['custom-field-map'];
+            $labels = CRM_Consentactivity_Service::customCheckboxFields();
+            $uFJoins = \Civi\Api4\UFJoin::get()
+                ->addSelect('uf_group_id')
+                ->addWhere('module', '=', 'CiviCampaign')
+                ->addWhere('entity_table', '=', 'civicrm_survey')
+                ->addWhere('entity_id', '=', $this->petition['id'])
+                ->setLimit(2)
```

Comment:
Why is a limit of 2 groups?

Reply:
Similar issue as the events. You can attach a contact and an activity profile for a petition.